### PR TITLE
ak_wh/hotfix_weights

### PR DESCRIPTION
**Pull request by willh727-OQC**
Fixing Acquire filter to handle floating points in a consistent way, to how its produced in calculating durations

### CODE REVIEW

**Comment by hamidelmaazouz** [comment]
@willh727-OQC consistency issues can be fixed by running `poetry run format-code`

File: src/qat/purr/compiler/instructions.py
Comment (hamidelmaazouz) [inline src/qat/purr/compiler/instructions.py:357]: similar processing as `XEngine.calculate_duration()` https://github.com/oqc-community/qat/blob/main/src/qat/purr/compiler/execution.py#L401
It's now an instance method, you could rip it off, make it a module utility function in `instructions.py` module and use it for your case:

```
calculate_duration(self)
```

Comment (willh727-OQC) [reply src/qat/purr/compiler/instructions.py:357]: Yeah good idea, I'll also replace it in calcaulte_duration so its done it one place

**Comment by elskysr** [comment]
recheck

File: src/qat/purr/compiler/instructions.py
Comment (hamidelmaazouz) [inline src/qat/purr/compiler/instructions.py:359]: put this in a variable `duration_nanos = calculate_duration(self, return_samples=False)` and reuse here

File: src/qat/purr/compiler/instructions.py
Comment (hamidelmaazouz) [inline src/qat/purr/compiler/instructions.py:693]: inline this function please, it's too short and I don't think it's used anywhere else

File: tests/qat/test_instructions.py
Comment (hamidelmaazouz) [inline tests/qat/test_instructions.py:177]: consider edge cases: negative, 0, some special positive value, ...

**Review by hamidelmaazouz — APPROVED** [review]
Love the work 😃  a few comments but I'm already happy. Also approving to unblock but please carry out the last changes I pointed to. Thanks for the fix @willh727-OQC

**Comment by willh727-OQC** [comment]
> Love the work 😃 a few comments but I'm already happy. Also approving to unblock but please carry out the last changes I pointed to. Thanks for the fix @willh727-OQC

Don't worry I can't merge it anyway!

File: src/qat/purr/compiler/instructions.py
Comment (willh727-OQC) [reply src/qat/purr/compiler/instructions.py:693]: I get your point! But I'm planning to use it in Whisqrs for ArbSeq! Just to make the removal of floating points consistent and the same as QAT.

Comment (hamidelmaazouz) [reply src/qat/purr/compiler/instructions.py:702]: the only thing worth sharing here is the rounding precision, which can be set properly via compiler config

cc/ @bgsach

Comment (hamidelmaazouz) [reply src/qat/purr/compiler/instructions.py:702]: It's okay for now, assume this function is gonna disappear soon, and that you need to pick the rounding precision from the config instead, I don't want to see apis leaking this way between qat and internal dependencies

**Review by hamidelmaazouz — APPROVED** [review]

File: src/qat/purr/compiler/instructions.py
Comment (willh727-OQC) [reply src/qat/purr/compiler/instructions.py:702]: Alright good to know theres a better solution going forward.

**Review by hamidelmaazouz — APPROVED** [review]

**Pull request merged**
